src/flint/fmpz.jl

```diff
@@ -641,17 +645,47 @@ div(x::Integer, y::ZZRingElem) = div(ZZRingElem(x), y)
 
 # Note Base.div is different to Nemo.div
 Base.div(x::Integer, y::ZZRingElem) = Base.div(ZZRingElem(x), y)
+Base.div(x::Integer, y::ZZRingElem, ::typeof(RoundToZero)) = Base.div(ZZ(x), y, RoundToZero)
+Base.div(x::Integer, y::ZZRingElem, ::typeof(RoundUp)) = Base.div(ZZ(x), y, RoundUp)
+Base.div(x::Integer, y::ZZRingElem, ::typeof(RoundDown)) = Base.div(ZZ(x), y, RoundDown)
 
 div(x::ZZRingElem, y::Integer) = div(x, ZZRingElem(y))
 
 # Note Base.div is different to Nemo.div
 Base.div(x::ZZRingElem, y::Integer) = Base.div(x, ZZRingElem(y))
+Base.div(x::ZZRingElem, y::Integer, ::typeof(RoundToZero)) = Base.div(x, ZZ(y), RoundToZero)
+Base.div(x::ZZRingElem, y::Integer, ::typeof(RoundUp)) = Base.div(x, ZZ(y), RoundUp)
+Base.div(x::ZZRingElem, y::Integer, ::typeof(RoundDown)) = Base.div(x, ZZ(y), RoundDown)
 
 divrem(x::ZZRingElem, y::Integer) = divrem(x, ZZRingElem(y))
 
 divrem(x::Integer, y::ZZRingElem) = divrem(ZZRingElem(x), y)
 
-Base.divrem(x::ZZRingElem, y::Int) = (Base.div(x, y), Base.rem(x, y))
+# Without the functions below, Julia defaults to `(div(x, y), rem(x, y))`
+Base.divrem(x::ZZRingElem, y::Integer) = (
+  Base.divrem(x, ZZ(y))
+)
```

Comment:
```suggestion
Base.divrem(x::ZZRingElem, y::Integer) = Base.divrem(x, ZZ(y))
```
why not make this oneline like the other ones